server/app.py

```diff
@@ -104,38 +181,105 @@ async def submit(request: SubmitRequest):
+        logger.info(f"got batch {data=}")
+        for tp, messages in data.items():
+            assert tp.topic == f"adala-output-{job_id}", f"Unexpected topic: {tp.topic}"
+            if messages:
+                handler(messages)
+            else:
+                logger.info(f"No messages in topic {tp.topic}")
+            if len(messages) > batch_size:
+                # assume we're done
+                logger.info(f"End of stream in topic {tp.topic}")
+                break
+    finally:
+        await consumer.stop()
+
+
+@app.post("/jobs/submit-streaming", response_model=Response[JobCreated])
+async def submit_streaming(request: SubmitStreamingRequest, background_tasks: fastapi.BackgroundTasks):
+    """
+    Submit a request to execute task `request.task_name` in celery.
 
-class JobStatusResponse(BaseModel):
+    Args:
+        request (SubmitStreamingRequest): The request model for submitting a job.
+
+    Returns:
+        Response[JobCreated]: The response model for a job created.
     """
-    Response model for getting the status of a job.
 
-    Attributes:
-        status (str): The status of the job.
-        processed_total (List[int]): The total number of processed records and the total number of records in job.
-            Example: [10, 100] means 10% of the completeness.
+    # TODO: get task by name, e.g. request.task_name
+    task = process_file_streaming
+    serialized_agent = pickle.dumps(request.agent)
+
+    logger.info(f"Submitting task {task.name} with agent {serialized_agent}")
+    result = task.delay(serialized_agent=serialized_agent)
+    # TODO pass batch size in request?
+    # TODO pass handler in request
+    dummy_handler = lambda batch: logger.info(f"Batch: {batch}")
+    background_tasks.add_task(poll_for_streaming_results, job_id=result.id, batch_size=2, handler=dummy_handler)
+    logger.info(f"Task {task.name} submitted with job_id {result.id}")
+
+    return Response[JobCreated](data=JobCreated(job_id=result.id))
+
+
+@app.post("/jobs/submit-batch", response_model=Response)
+async def submit_batch(batch: BatchData):
     """
+    Submits a batch of data to an existing streaming job.
+    Will push the batch of data into Kafka in a topic specific to the job ID
 
-    status: Status
-    # processed_total: List[int] = Annotated[List[int], AfterValidator(lambda x: len(x) == 2)]
+    Args:
+        batch (BatchData): The data to push to Kafka queue to be processed by agent.arun()
 
-    class Config:
-        use_enum_values = True
+    Returns:
+        Response: Generic response indicating status of request
+    """
+
+    topic = f"adala-input-{batch.job_id}"
+    producer = AIOKafkaProducer(
+        bootstrap_servers=settings.kafka_bootstrap_servers,
+        value_serializer=lambda v: json.dumps(v).encode("utf-8"),
+    )
+    await producer.start()
+
+    try:
+        for record in batch.data:
+            await producer.send_and_wait(topic, value=record)
```

Comment:
do we have to do this one record at a time? I'd guess there's a way to send this in batches, but maybe aiokafka doesn't support this?